dropbox/dropbox_client.py

```diff
@@ -361,6 +361,9 @@ def check_and_refresh_access_token(self):
         needs_refresh = self._oauth2_access_token_expiration and \
             (datetime.utcnow() + timedelta(seconds=TOKEN_EXPIRATION_BUFFER)) >= \
             self._oauth2_access_token_expiration
+        # additional check for refreshing token without having above logic get too messy
+        needs_refresh = needs_refresh or (self._oauth2_refresh_token and
```

Comment:
can we not just add an or `not oauth2_expiration_time` rather than an entire new line?

Reply:
Ok I think I fixed it